### PR TITLE
fix disabling ALL_METRICS does not work as expected

### PR DESCRIPTION
**Pull request by zirain**
Signed-off-by: hejianpeng <hejianpeng2@huawei.com>

**Please provide a description of this PR:**

fixes: #43178

### CODE REVIEW

File: pilot/pkg/model/telemetry.go [hunk 1]
```diff
@@ -1021,31 +1043,47 @@ var metricToPrometheusMetric = map[string]string{
...
-			Name:       metricName,
-			Drop:       override.Disabled,
-		}
-		for _, t := range override.Tags {
-			if t.Remove {
-				mc.TagsToRemove = append(mc.TagsToRemove, t.Name)
-			} else {
-				mc.Dimensions[t.Name] = t.Value
+
+	metricCfg := filterConfig.MetricsForClass(class)
```
Comment (kyessenov) [inline pilot/pkg/model/telemetry.go:1063]: reuse variable `listenerCfg` above

File: pilot/pkg/model/telemetry.go [hunk 2]
```diff
@@ -1021,31 +1043,47 @@ var metricToPrometheusMetric = map[string]string{
...
+	if !metricCfg.Disabled {
```
Comment (kyessenov) [inline pilot/pkg/model/telemetry.go:1064]: superfluous check

**Review by kyessenov — CHANGES_REQUESTED** [review]
LGTM, some style improvement.

File: releasenotes/notes/43179.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: telemetry
+issues:
+  - 43178
+releaseNotes:
+  - |
+    **Fixed** disabling ALL_METRICS does not work as expected.
```
Comment (GregHanson) [inline releasenotes/notes/43179.yaml:8]:
```suggestion
    **Fixed**  an issue where `ALL_METRICS` does not disable metrics as expected.
```

**Review by kyessenov — APPROVED** [review]
I'd like @lei-tang to take a look since we're close to a release.

**Review by lei-tang — APPROVED** [review]

**Review by GregHanson — APPROVED** [review]
release note looks good

**Review by ericvn — APPROVED** [review]
Test and RN look good.

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**